src/test/java/com/rocketden/main/service/ProblemServiceTests.java

```diff
@@ -32,9 +37,45 @@
     @InjectMocks
     private ProblemService problemService;
 
+    private static final int ID = 10;
     private static final String NAME = "Sort a List";
     private static final String DESCRIPTION = "Sort the given list in O(n log n) time.";
 
+    private static final String INPUT = "[1, 2, 8]";
+    private static final String OUTPUT = "8";
+
+    @Test
+    public void getProblemSuccess() {
+        Problem expected = new Problem();
+        expected.setId(ID);
+        expected.setName(NAME);
+        expected.setDescription(DESCRIPTION);
+
+        ProblemTestCase testCase = new ProblemTestCase();
+        testCase.setInput(INPUT);
+        testCase.setOutput(OUTPUT);
+        expected.addTestCase(testCase);
+
+        Mockito.doReturn(Optional.of(expected)).when(repository).findById(ID);
```

Comment:
What is the use of `Optional` here? It might be useful to add a comment for this?

Reply:
`findById` returns an Optional as a wrapper around the actual object for whatever reason rather than just returning null when not found. Kinda weird I agree but yeah I added a comment to clarify this

Reply:
Huh interesting -- any idea why this is different than the other "findByX" methods?

Reply:
Not sure tbh haha - [this](https://stackoverflow.com/questions/60638660/why-findbyid-returns-optional) goes into some of the use cases tho

Reply:
Oh interesting, there's a similar concept in OCaml. Ok yeah no problem, this looks good.